server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -842,6 +859,12 @@ public final CommitStats commitStats() {
      */
     public abstract long getPersistedLocalCheckpoint();
 
+    /**
+     * @return the latest checkpoint that has been processed but not necessarily persisted.
+     * Also see {@link #getPersistedLocalCheckpoint()}
+     */
+    public abstract long getProcessedLocalCheckpoint();
```

Comment:
@satyajitg28  This is breaking one, we need to provide default implementation

Reply:
Fixed.